### ISSUE TITLE
systemd journal support for linux?

### ISSUE THREAD

**Issue by ahjolinna** [alamminsalo/orion#101]
currently logging/debugging is little bit lacking in general so it would be nice to have something like "journald" support on linux, on other platfom I dont know what they use or do.
also I know that there are few Qt based video players have done this, like [VLC](http://git.videolan.org/?p=vlc.git;a=commit;h=15ed09825868776d03e5891a922b28edd9d61f4b&utm_source=anzwix) and [bomi](https://github.com/xylosper/bomi/commit/2494d806bb84078e8416a6d775becfc85b19da5e)

why is this useful (for linux users)?
- single place to look for log output 
- no need to handle log file creation/rotation/collisions/running out of space
     - just send it to the Journal interface and let it handle the rest
- the Journal API supports priorities and other metadata
     -  users can see right away which messages are critical/non critical due to the build in journalctl message highlighting
- powerful filtering & cross reference
     -  users can filter out just the orion log output or check what happened at the same time as the orion message was logged (orion error accompanied by a x server error, etc.)

- powerful time based lookup - users can check all orion log messages from last run, last day, last hour, etc.

More Journal resources:
http://www.freedesktop.org/software/systemd/man/journalctl.html
http://0pointer.de/blog/projects/journal-submit.html

------

- add options to select logging level or turn it off for stdout, stderr, journal and external file
- if command line option --log-level is given, this will override logging level for stdout only

By default, orion will log with journal only.
stdout/stderr is disabled by default because orion is GUI application and usually, it will be launched without terminal screen which results into discarding all stdout/stderr outputs.

Also, if someone gave a command line option, it means he or she executed orion from terminal, so I think it makes sense that --log-level affects stdout only.
Since this option already exists, I don't think additional env. var. is needed.

What I'm still concerning is default logging level(verbosity).
In fact, because of my lazyness, orion doesn't log so much things yet.
But I believe journald is fast enough to raise the default logging level for journal to 'debug' because 'debug' can show much more information on crash.

**Comment by rakslice** [comment]
Do you have some idea of what you would want to go in the log, compared with the current debug output?

**Comment by ahjolinna** [comment]
@rakslice : well more debug options the better, this is what bomi-player has done for example (also Qt5/qml based and uses MPV + its crossplatfom): 
![image](https://cloud.githubusercontent.com/assets/6993072/26138173/d0752056-3acf-11e7-9760-0c7fe854dd5e.png)
and other, not just systemd:
![image](https://cloud.githubusercontent.com/assets/6993072/26138179/e1a2a15a-3acf-11e7-8af5-96903e1afff0.png)
the "viewer" is this: 

![image](https://cloud.githubusercontent.com/assets/6993072/26138203/0a31fb52-3ad0-11e7-9fb3-af20f92eacb7.png)

**Comment by rakslice** [comment]
Note that Qt supports redirecting log output to systemd as a build-time option. https://www.ics.com/blog/whats-new-qt-560-logging-syslog-and-journald.  As others have noted, this is mainly intended as a measure for embedded systems and the like where there is no useful access to filesystem-based logs.

Putting log output to a central place makes sense for operational output of a service, but not for debug output of an application, where the use case is that it is almost always turned off [and ideally not compiled in ;) ] for performance reasons, and the when you turn it on you actually want to see it right away while the program is running.

**Comment by ahjolinna** [comment]
I have never seen any performance issues with bomi, mpc-qt or vlc. Also it can be disabled by default in the settings anyways...and if someone notice performance issues with this, I think that I would be a good time to upgrade your system or its really broken (I would argue that its the same thing)

I think it would be good to have the same option what bomi/mpc-qt has, and maybe even add that log viewer for those who want it (novice users ?)

why I like the same implementation what bomi/mpc-qt has?
well it would make bug reporting easier and faster for those novice users who don't know how to do basic debugging.

also if there is an issue that is hard to replicate, then this would help A LOT (like #167 )

**Comment by IdleGandalf** [comment]
I agree with @rakslice here, systemd's journal is not the place where debug output should go. Starting a program with -d/--debug or -v for more verbose output is wide spread and every unix user should be aware of it, so implementing a proper debug/verbose flag should be preferred in any case.

Just my two cents.
